src/Datagrid/Datagrid.php

```
@@ -330,6 +291,64 @@ public function getPaginationParameters(int $page): array
         return ['filter' => $values];
     }
 
+    private function applyFilters(array $data): void
+    {
+        foreach ($this->getFilters() as $name => $filter) {
+            $this->values[$name] = $this->values[$name] ?? null;
+            $filterFormName = $filter->getFormName();
+            if (isset($this->values[$filterFormName]['type'], $this->values[$filterFormName]['value']) &&
+                ('' !== $this->values[$filterFormName]['type'] || '' !== $this->values[$filterFormName]['value'])
+            ) {
+                $filter->apply($this->query, $data[$filterFormName]);
+            }
+        }
+    }
+
+    private function applySorting(): void
+    {
+        if (isset($this->values['_sort_by'])) {
```

Can you invert this check to fail fast

---

Like this?